libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -823,16 +823,26 @@ bool ASTJsonConverter::visit(Literal const& _node)
 	if (!util::validateUTF8(_node.value()))
 		value = Json::nullValue;
 	Token subdenomination = Token(_node.subDenomination());
+
+	Json::Value subdenominationJson;
+
+	switch (subdenomination)
+	{
+		case Token::Illegal:
+			subdenominationJson = Json::nullValue;
+			break;
+		case Token::Identifier:
```

Comment:
You should just emit the appropriate token in the parser and not have this extra case here.

Reply:
But you said yourself that we can't add an extra token...